src/firejail/profile.c

```diff
@@ -1045,6 +1045,90 @@ int profile_check_line(char *ptr, int lineno, const char *fname) {
 		return 0;
 	}
 
+#ifdef HAVE_LANDLOCK
+	// Landlock ruleset paths
+	if (strcmp(ptr, "landlock") == 0) {
+		if (arg_landlock == -1) arg_landlock = create_full_ruleset();
+		const char *home_dir = env_get("HOME");
+		int home_fd = open(home_dir,O_PATH | O_CLOEXEC);
+		struct landlock_path_beneath_attr target;
+		target.parent_fd = home_fd;
+		target.allowed_access = LANDLOCK_ACCESS_FS_READ_FILE | LANDLOCK_ACCESS_FS_READ_DIR | LANDLOCK_ACCESS_FS_WRITE_FILE | LANDLOCK_ACCESS_FS_REMOVE_FILE | LANDLOCK_ACCESS_FS_REMOVE_DIR | LANDLOCK_ACCESS_FS_MAKE_CHAR | LANDLOCK_ACCESS_FS_MAKE_DIR | LANDLOCK_ACCESS_FS_MAKE_REG | LANDLOCK_ACCESS_FS_MAKE_SYM;
+		if (landlock_add_rule(arg_landlock,LANDLOCK_RULE_PATH_BENEATH,&target,0)) {
+			fprintf(stderr,"An error has occured while adding a rule to the Landlock ruleset.\n");
+		}
+		close(home_fd);
+		if (add_read_access_rule_by_path(arg_landlock, "/bin/")) {
+			fprintf(stderr,"An error has occured while adding a rule to the Landlock ruleset.\n");
+		}
+		if (add_execute_rule_by_path(arg_landlock, "/bin/")) {
+			fprintf(stderr,"An error has occured while adding a rule to the Landlock ruleset.\n");
+		}
+		if (add_read_access_rule_by_path(arg_landlock, "/dev/")) {
+			fprintf(stderr,"An error has occured while adding a rule to the Landlock ruleset.\n");
+		}
+		if (add_read_access_rule_by_path(arg_landlock, "/etc/")) {
+			fprintf(stderr,"An error has occured while adding a rule to the Landlock ruleset.\n");
+		}
+		if (add_read_access_rule_by_path(arg_landlock, "/lib/")) {
+			fprintf(stderr,"An error has occured while adding a rule to the Landlock ruleset.\n");
+		}
+		if (add_execute_rule_by_path(arg_landlock, "/lib/")) {
+			fprintf(stderr,"An error has occured while adding a rule to the Landlock ruleset.\n");
+		}
+		if (add_read_access_rule_by_path(arg_landlock, "/opt/")) {
+			fprintf(stderr,"An error has occured while adding a rule to the Landlock ruleset.\n");
+		}
+		if (add_execute_rule_by_path(arg_landlock, "/opt/")) {
+			fprintf(stderr,"An error has occured while adding a rule to the Landlock ruleset.\n");
+		}
+		if (add_read_access_rule_by_path(arg_landlock, "/usr/")) {
+			fprintf(stderr,"An error has occured while adding a rule to the Landlock ruleset.\n");
+		}
+		if (add_execute_rule_by_path(arg_landlock, "/usr/")) {
+			fprintf(stderr,"An error has occured while adding a rule to the Landlock ruleset.\n");
+		}
+		if (add_read_access_rule_by_path(arg_landlock, "/var/")) {
+			fprintf(stderr,"An error has occured while adding a rule to the Landlock ruleset.\n");
+		}
+		return 0;
+	}
+	if (strncmp(ptr, "landlock.proc ", 14) == 0) {
+			if (strncmp(ptr+14, "no", 2) == 0) arg_landlock_proc = 0;
+			else if (strncmp(ptr+14, "ro", 2) == 0) arg_landlock_proc = 1;
+			else if (strncmp(ptr+14, "rw", 2) == 0) arg_landlock_proc = 2;
+			return 0;
+		}
```

Comment:
From my experience with crablock I can tell that `--landlock.path_beneath="/proc:READ_FILE"` is often enough. Maybe it would be interesting to expose this as a read-only-files-but-no-dirs option.